src/cmiss_petsc.f90

```diff
@@ -1156,6 +1155,12 @@ SUBROUTINE TSSetFromOptions(ts,ierr)
       PetscInt ierr
     END SUBROUTINE TSSetFromOptions
 
+    SUBROUTINE TSSetExactFinalTime(ts,eftopt,ierr)
+      TS ts
+      PetscBool eftopt
```

Comment:
The eftopt is not a PetscBool. According to http://www.mcs.anl.gov/petsc/petsc-current/docs/manualpages/TS/TSSetExactFinalTime.html the option should be TSExactFinalTimeOption. The various enums for TSExactFinalTimeOption will need to be set as parameters.